### PR TITLE
ci: ci update

### PR DESCRIPTION
**Pull request by xtx1130**
update gitlab ci for koa repo

### CODE REVIEW

**Comment by xtx1130** [comment]
Is anyone here for code review?

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -2,18 +2,23 @@ name: Node.js CI
...
 
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
 
     strategy:
       matrix:
         node-version: [12.x, 14.x, 16.x]
```
Comment (3imed-jaberi) [inline .github/workflows/node.js.yml:20]: I think soon we will drop the 12.x version and we will support 14+ or maybe drop 14.x also and support 16+!!

**Review by 3imed-jaberi — APPROVED** [review]
+1

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (xtx1130) [reply .github/workflows/node.js.yml:20]: Yes, I think when 12.x is outdated, we can add 17.x in ci

Comment (miwnwski) [reply .github/workflows/node.js.yml:20]: I don't think Koa 2 will drop any major version support. If we're following semver that would be a major break.

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -2,18 +2,23 @@ name: Node.js CI
...
+        os: [macos-latest, ubuntu-latest, windows-latest]
```
Comment (miwnwski) [inline .github/workflows/node.js.yml:21]: I ask myself, is this really necessary? :) We're not compiling anything, if node works on the platform, Koa will too. Just seems like more work for no real benefit? Do you have a different opinion here or am I missing something maybe?

**Comment by miwnwski** [comment]
I'll close this unless you feel strongly about this or I'm missing something. Don't hesitate to reopen but right now I don't see the need to test Koa on multiple platforms.

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (xtx1130) [reply .github/workflows/node.js.yml:20]: Node v12.x is END-OF-LIFE now, and node v18.x will be active LTS in 2022/10/18. I think it is necessary to update ci version to consistent with nodejs releases

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (xtx1130) [reply .github/workflows/node.js.yml:21]: Nodejs has some APIs that return inconsistent results between win and other platforms(such as path). I think add these platforms can prevent some cross-platform bugs from happening.

**Pull request closed without merging**